parity/migration.rs

```diff
@@ -215,6 +219,12 @@ fn exists(path: &Path) -> bool {
 	fs::metadata(path).is_ok()
 }
 
+// in-place upgrades that do nothing when called repeatedly
+fn run_inplace_upgrades(path: &Path) -> Result<(), Error> {
+	try!(migrations::upgrade_account_bloom(path));
+	Ok(())
```

Comment:
I don't think this strategy will work once we add more migrations on top of this one, since we need this account bloom migration to happen before others which may affect it.

Why not change `Migration` to supply `Arc<Database>` as `source`, and copy everything over for each column, and then additionally if column == COL_STATE, you also do the same stuff as `upgrade_account_bloom` would?

I think this would also require backporting my `pre_columns` fixes from master as consolidated db migrations are currently broken.